pkg/tlsx/clients/clients.go

```diff
@@ -207,3 +212,33 @@ func IsSelfSigned(authorityKeyID, subjectKeyID []byte) bool {
 	}
 	return false
 }
+
+// IsMisMatchedCert returns true if cert names(subject common name + alternative names) does not contain host
+func IsMisMatchedCert(host string, names []string) bool {
+	hostTokens := strings.Split(host, ".")
+	for _, name := range names {
+		// if no wildcard, retrun false if name matches the host
+		if !strings.Contains(name, "*") {
+			if strings.EqualFold(name, host) {
+				return false
+			}
+		} else {
+			matched := false
+			nameTokens := strings.Split(name, ".")
+			if len(hostTokens) == len(nameTokens) {
+				for i, token := range nameTokens {
```

Comment:
Wouldn't something like this have the same result?
```go
for i, token := range nameTokens {
    if stringsutil.EqualFoldAny(token, "*", hostTokens[i]) {
        return false
    }
}
```

Reply:
yes, we can utilize `stringsutil.EqualFoldAny` but can't return false on single token match.